### PR TITLE
Teacher student prem

### PR DESCRIPTION
**Pull request by karinocheretny**
-new isTeacher attribute in Users table
-new checkbox in sign up for isTeacher
-serve separate landing pages based on attribute
-Adjusted database (migration and seeding) to pass with the new attributes

### CODE REVIEW

**Comment by Durbin-Elijah** [comment]
The changes in frontend look good, I made sure the changes in my PR redesign-react are compatible, but this one should be done first.

I can't approve just yet because I don't know about the changes to the backend and testing, so @Justin-Fernbaugh  would probably be the best person to review those areas.

File: server/app/api/courses.js
Comment (nilsstreedain) [inline server/app/api/courses.js:1]: Why is this removed? Are we now checking if isTeacher is true, or are we allowing this now?

File: server/db/seeders/20230202033013-add_api_test_data.js [hunk 1]
```diff
@@ -76,11 +79,11 @@ module.exports = {
       role: 'student'
     }], {})
 
-    let comboenrollmentsection1course2 = await queryInterface.bulkInsert('Enrollments', [{
-      userId: comboUser,
-      sectionId: section1course1,
-      role: 'student'
-    }], {})
+    //let comboenrollmentsection1course2 = await queryInterface.bulkInsert('Enrollments', [{
```
Comment (nilsstreedain) [inline server/db/seeders/20230202033013-add_api_test_data.js:82]: Maybe don't leave these as comments? Can this be removed or is it being kept to rework?

**Review by nilsstreedain — CHANGES_REQUESTED** [review]
Added a few comments with questions ~~but we can push it through if you think no changes are necessary, just don't want something to slip through.~~

Edit: Looks like tests are failing, should take a look at that

File: server/app/api/courses.js
Comment (nilsstreedain) [reply server/app/api/courses.js:1]: *Forgot to link it to the line, was referring to lines 98-100 that got removed

**Comment by Justin-Fernbaugh** [comment]
Were you able to take a look at the tests on the backend? I'm getting 219 failed tests @karinocheretny

**Comment by nilsstreedain** [comment]
> Were you able to take a look at the tests on the backend? I'm getting 219 failed tests @karinocheretny

Shoot yeah, I must have done something wrong when testing, I'm getting the same thing

File: server/app/api/courses.js
Comment (karinocheretny) [reply server/app/api/courses.js:1]: Not sure if I mentioned but the databases need to be re-migrated and re-seeded. Did you do that by chance? the tests should be passing with those changes

File: server/db/seeders/20230202033013-add_api_test_data.js (hunk 1 above)
Comment (karinocheretny) [reply server/db/seeders/20230202033013-add_api_test_data.js:82]: Commented it out in case we ever want a combo user again? I can remove for now

Comment (nilsstreedain) [reply server/db/seeders/20230202033013-add_api_test_data.js:82]: Imo its best practice to remove them and then use git history if we want to bring something back but either way is fine here

File: server/app/api/courses.js
Comment (nilsstreedain) [reply server/app/api/courses.js:1]: I think you may have meant to respond in the overall PR log (yep did that by mistake lol)? This comment was specifically on the removal of lines 98-100 that checked to make sure a teacher wouldn't join their own course. I couldn't code replacing this anywhere but I may have missed it

**Comment by Durbin-Elijah** [comment]
When resolving conflicts with Signup.js please prioritize dev, it should work as you intend and keep the functionality I added.

File: server/app/api/courses.js
Comment (karinocheretny) [reply server/app/api/courses.js:1]: this check It's not necessary anymore due to the new architecture of the split home pages (teachers don't have an option to join courses), so if a user joins a course we can just assume it is a student as they're the only ones with access to that functionality

**Review by nilsstreedain — APPROVED** [review]
Looks great, good work!
